packages/tupaia-web/src/pages/Register/Register.tsx

```diff
@@ -0,0 +1,174 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+import React from 'react';
+import styled from 'styled-components';
+import { useForm } from 'react-hook-form';
+import Typography from '@material-ui/core/Typography';
+import { useRegister } from '../../api/mutations';
+import { SignupComplete } from './SignUpComplete';
+import { SubmitHandler } from 'react-hook-form';
+import { TextField, AuthModalBody, AuthModalButton } from '../../components';
+import { FORM_FIELD_VALIDATION } from '../../constants';
+import { Checkbox } from '@tupaia/ui-components';
+import { Link } from 'react-router-dom';
+import { MODAL_ROUTES } from '../../constants';
+
+const ModalBody = styled(AuthModalBody)`
+  width: 53rem;
+`;
+
+const StyledForm = styled.form`
+  margin-top: 1rem;
+  width: 42rem;
+  max-width: 100%;
+
+  @media (min-width: ${({ theme }) => theme.breakpoints.values.sm}px) {
+    display: grid;
+    grid-template-columns: 1fr 1fr;
+    column-gap: 2rem;
+    row-gap: 0;
+  }
+`;
+
+const StyledCheckbox = styled(Checkbox)`
+  display: flex;
+`;
+
+const LinkText = styled(Typography)`
+  font-weight: 400;
+  font-size: 11px;
+  line-height: 15px;
+  color: white;
+
+  a {
+    color: white;
+  }
+
+  ${AuthModalButton} + & {
+    margin-top: 1.3rem;
+  }
+`;
+
+const FullWidthColumn = styled.div`
+  grid-column: 1/-1;
+`;
+
+export const Register = () => {
+  const { mutate: onSubmit, isLoading, isSuccess, isError, error } = useRegister();
+  const { errors, handleSubmit, getValues, register } = useForm();
+
+  return (
+    <ModalBody
+      title="Register"
+      subtitle={!isSuccess ? 'Enter your details below to create an account' : undefined}
+    >
+      {isSuccess ? (
+        <SignupComplete />
+      ) : (
+        <>
+          {isError && <Typography color="error">{error.message}</Typography>}
+          <StyledForm onSubmit={handleSubmit(onSubmit as SubmitHandler<any>)} noValidate>
+            <TextField
+              name="firstName"
+              label="First name *"
```

Comment:
What do you think about a way to handle the `*` in the label dynamically, when `required` is true? 
Also might be worth passing the `required` prop to the `TextField` (or does the `inputRef` apply this to the input as well?)

Reply:
I ended up adding a form component with context and moving some of this repeated code to the TextField component. thanks for the suggestion.